### PR TITLE
Closes #88. Mapreduce streaming using spark pipe

### LINKED ISSUE #88: Spark cloner example using hadoop streaming

### PR DESCRIPTION
**Pull request by madryk**

### CODE REVIEW

File: iis-core/src/main/java/eu/dnetlib/iis/core/spark/pipe/SparkPipeMapReduce.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+ */
+public class SparkPipeMapReduce {
+	
+	
+	//------------------------ LOGIC --------------------------
+	
+	public static void main(String[] args) throws IOException, ClassNotFoundException {
+		
+		
+        SparkPipeMapReduceParameters params = parseParameters(args);
```
Comment (lukdumi) [inline iis-core/src/main/java/eu/dnetlib/iis/core/spark/pipe/SparkPipeMapReduce.java]: Improper indentation

File: iis-core/src/main/java/eu/dnetlib/iis/core/spark/pipe/SparkPipeExecutor.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ */
+public class SparkPipeExecutor implements Serializable {
+
+	private static final long serialVersionUID = 1L;
+
+
+	//------------------------ LOGIC --------------------------
+	
+	/**
+	 * Imitates map part of hadoop streaming job
```
Comment (lukdumi) [inline iis-core/src/main/java/eu/dnetlib/iis/core/spark/pipe/SparkPipeExecutor.java]: The comment does not mean much for people who do not know hadoop streaming. And even those would not know that this method assumes that the invoked script must return lines containing tab separated key/values and takes and operates on keys.

File: iis-core/src/main/java/eu/dnetlib/iis/core/spark/pipe/SparkPipeExecutor.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+					String[] splittedPair = line.split("\t");
+					return new Tuple2<String, String>(splittedPair[0], (splittedPair.length == 1) ? null : splittedPair[1]);
+				});
+
+		return outputRecords;
+	}
+
+	/**
+	 * Imitates reduce part of hadoop streaming job
+	 */
```
Comment (lukdumi) [inline iis-core/src/main/java/eu/dnetlib/iis/core/spark/pipe/SparkPipeExecutor.java]: The same here, the comment is too laconic.

File: iis-core/src/main/java/eu/dnetlib/iis/core/spark/pipe/SparkPipeMapReduce.java (hunk 1 above)
Comment (madryk) [inline iis-core/src/main/java/eu/dnetlib/iis/core/spark/pipe/SparkPipeMapReduce.java]: :+1:

File: iis-core/src/main/java/eu/dnetlib/iis/core/spark/pipe/SparkPipeExecutor.java (hunk 2 above)
Comment (madryk) [inline iis-core/src/main/java/eu/dnetlib/iis/core/spark/pipe/SparkPipeExecutor.java]: I added some more detailed comment

File: iis-core/src/main/java/eu/dnetlib/iis/core/spark/pipe/SparkPipeExecutor.java (hunk 3 above)
Comment (madryk) [inline iis-core/src/main/java/eu/dnetlib/iis/core/spark/pipe/SparkPipeExecutor.java]: Added some more detailed comment

**Comment by madryk** [comment]
I also added unit test for SparkPipeExecutor

**Pull request closed without merging**